### PR TITLE
Nullary variadic

### PR DESCRIPTION
**Pull request by gramalingam**
The existing interpretation of a variadic parameter is that it represents 1 or more parameters. However, for a Loop we may have zero or more loop-carried-state parameters. This PR adds an extension to support such ops that may have a variadic parameter representing zero or more parameters. (The PR actually adds an ability to specify a minimum arity, which defaults to 1, which is the original behavior.)

### CODE REVIEW

**Comment by gramalingam** [comment]
@houseroad any feedback on this? We could also do this by adding another value in the enum FormalParameterOption (say "ZeroOrMore"). The current form allows variadic ops that require a minimum number of parameters (other than 0 or 1) as well, in case we need it in the future.

File: onnx/defs/operator_sets.h [hunk 1]
```diff
@@ -533,6 +533,7 @@ class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, IsInf);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, NonMaxSuppression);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, ReverseSequence);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, RoiAlign);
+class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, Loop);
```
Comment (houseroad) [inline onnx/defs/operator_sets.h]: This should be opset 11, since we already release 1.5. 10 should be frozen.

File: onnx/defs/operator_sets.h [hunk 2]
```diff
@@ -568,14 +564,13 @@ class OpSet_Onnx_ver10 {
...
-           Onnx, 10, IsInf)>());
+    fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, IsInf)>());
     fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(
            Onnx, 10, NonMaxSuppression)>());
     fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(
            Onnx, 10, ReverseSequence)>());
-     fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(
-            Onnx, 10, RoiAlign)>());
+    fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, RoiAlign)>());
+    fn(GetOpSchema<ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 10, Loop)>());
```
Comment (houseroad) [inline onnx/defs/operator_sets.h]: same here.

File: onnx/defs/controlflow/defs.cc [hunk 3]
```diff
@@ -529,9 +529,9 @@ point-wise operators (e.g. dropout, residual connections, linear layer).
 
 ONNX_OPERATOR_SET_SCHEMA(
     Loop,
-    1,
+    10,
```
Comment (houseroad) [inline onnx/defs/controlflow/defs.cc]: here, bump up please

File: onnx/defs/operator_sets.h (hunk 1 above)
Comment (gramalingam) [reply onnx/defs/operator_sets.h]: Thanks for catching this. I have updated it. (I created this PR a while back when it was still opset 10 and forgot about it now.)

**Comment by gramalingam** [comment]
@houseroad @linkerzhang : any other comments on this PR? Thanks!

**Pull request merged**